### PR TITLE
Add FDTD test.

### PR DESCRIPTION
**Pull request by samcunliffe**
Adds an example FDTD test (current functionality). Not necessarily meaningful, but a run of the example scripts with FD compile flag. This at least tests against regression.

Also runs the unit tests with the FD flag in the CI for BUILD_TESTING builds. And update the documentation.

***
### Dependencies

- #121 
- #119

### CODE REVIEW

File: .github/workflows/linux_tests.yml [hunk 1]
```diff
@@ -70,9 +77,19 @@ jobs:
...
-          py.test ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x
+          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x  -m "not fdtd_build_only"
+
+      - name: Run FDTD TDMS system tests
+        if: matrix.derivative_type == 'FD'
+        working-directory: ${{runner.workspace}}/build
+        shell: bash
+        run: |
+          export OMP_NUM_THREADS=1
+          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x  -m fdtd_build_only
```
Comment (samcunliffe) [inline .github/workflows/linux_tests.yml:95]:
```suggestion
          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x  -m fdtd_build_only
          
```

File: .github/workflows/windows_tests.yml [hunk 2]
```diff
@@ -74,11 +81,23 @@ jobs:
...
         working-directory: ${{runner.workspace}}/build
         shell: bash
         run: |
           export OMP_NUM_THREADS=1
           export PATH=$(dirname "$(which MATLAB)")/win64/:$PATH
           export PATH=/c/Miniconda/envs/test/Library/bin/:$PATH
-          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/test_cli.py -s -x
+          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/test_cli.py -s -x -m "not fdtd_build_only"
+
+      - name: Run PSTD TDMS system tests
```
Comment (samcunliffe) [inline .github/workflows/windows_tests.yml:95]:
```suggestion
      - name: Run FDTD TDMS system tests
```

File: .github/workflows/linux_tests.yml [hunk 3]
```diff
@@ -21,6 +21,10 @@ jobs:
         build_testing:
           - ON
           - OFF
+        # TODO: remove this once MATLAB dependency is removed and mexFunction is replaced
```
Comment (willGraham01) [inline .github/workflows/linux_tests.yml:24]: Worth breaking this out into another issue, also mentioning that we should turn this option from a compile-time option to a command-line option.

File: tdms/tests/system/test_fdtd.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+def test_fs():
+    """
+    Ensure that the free space FDTD output matches the reference. Checks
+    that the output .mat file (with a HDF5 format) contains tensors with
+    relative mean square values within numerical precision of the reference.
+
+    Note: At the moment (due to MATLAB) this test only works if tdms is compiled with
+    """
+
+    # misleading filename here
```
Comment (willGraham01) [inline tdms/tests/system/test_fdtd.py:28]: Agreed - can we (or get Peter to if he's the only one with access) get these named a bit more consistently? (`fdtd` over `pstd` of course)

Comment (samcunliffe) [reply tdms/tests/system/test_fdtd.py:28]: It's me.
We can, or we wait for peter to give us a meaningful test before re-uploading.

Comment (willGraham01) [reply tdms/tests/system/test_fdtd.py:28]: Let's leave it as it is for now then, and when we get a more meaningful test update the names at the same time as we re-upload

Comment (samcunliffe) [reply tdms/tests/system/test_fdtd.py:28]: Yeah there is also
* #116

File: .github/workflows/linux_tests.yml [hunk 5]
```diff
@@ -92,4 +92,5 @@ jobs:
         shell: bash
         run: |
           export OMP_NUM_THREADS=1
-          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x  -m fdtd_build_only
\ No newline at end of file
+          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x  -m fdtd_build_only
+
```
Comment (giordano) [inline .github/workflows/linux_tests.yml:96]:
```suggestion

```

Comment (samcunliffe) [reply .github/workflows/linux_tests.yml:96]: Oh thanks. Something's wrong with my vim config for yml.

**Review by willGraham01 — APPROVED** [review]
Looks good. The fact that we have to recompile the executable (and thus have almost doubled the number of workflows we now have) makes me a little sad, but necessary until we can make this a command line option.

File: .github/workflows/windows_tests.yml [hunk 6]
```diff
@@ -80,11 +87,17 @@ jobs:
...
+        # or 'FD', and filter the pytest labels accordingly
+        # pytest <blah> -m fdtd_build_only
         working-directory: ${{runner.workspace}}/build
         shell: bash
         run: |
           export OMP_NUM_THREADS=1
           export PATH=$(dirname "$(which MATLAB)")/win64/:$PATH
           export PATH=/c/Miniconda/envs/test/Library/bin/:$PATH
-          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/test_cli.py -s -x
+          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/test_cli.py -s -x
```
Comment (giordano) [inline .github/workflows/windows_tests.yml:103]:
```suggestion
          pytest ${GITHUB_WORKSPACE}/tdms/tests/system/test_cli.py -s -x
```

**Pull request merged**